### PR TITLE
Reject duplicate prefixes in demerit parser

### LINKED ISSUE #473: [PE-D][Tester C] [Bug] `demerit` accepts duplicate key prefixes (e.g., `i=`/`di=`) without rejecting ambiguity
### Description
`demerit` appears to allow duplicate targeting/rule prefixes and proceeds, instead of rejecting ambiguous command structure.

### Steps to Reproduce
1. Ensure resident `A1234567X` exists. [ASSUMPTION]
2. Run: `demerit i=A1234567X i=A7654321X di=18 di=1`
3. Observe whether command is rejected or proceeds with one set of values.

### Expected Result
Duplicate `i=` or `di=` prefixes should be rejected as invalid/ambiguous input.

### Actual Result
Command proceeds using the latest provided values for each prefix respectively.

### Severity/Type Justification
- Type: `type.FunctionalityBug`
- Severity: `severity.Low`
- Reasoning: Ambiguous input handling is weak and can lead to unintended application.

#### Result feedback screenshot after running `demerit i=A1234567X i=A7654321X di=18 di=1`
Image

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by kenghow**
Fixes #473 ambiguous demerit inputs by rejecting repeated `i=` and `di=` prefixes.

Why:
- `demerit` should target exactly one student and one rule index
- repeated prefixes make the command ambiguous
- current behavior silently uses the last value

What changed:
- reject repeated `i=` in `DemeritCommandParser`
- reject repeated `di=` in `DemeritCommandParser`
- add parser tests for duplicate `i=`, duplicate `di=`, and both duplicated

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/DemeritCommandParser.java [hunk 1]
```diff
@@ -36,4 +38,16 @@ public DemeritCommand parse(String args) throws ParseException {
 
         return new DemeritCommand(studentId, ruleIndex, remark);
     }
+
+    /**
+     * Returns true if the given prefix appears more than once in the raw argument string.
+     */
+    private boolean hasDuplicatePrefix(String args, String prefixString) {
```
Comment (pavithra-achar) [inline src/main/java/seedu/address/logic/parser/DemeritCommandParser.java]: Hi, can I ask why you have defined a new method for this when `verifyNoDuplicatePrefixesFor` exists in `ArgumentMultiMap`?

Comment (kenghow) [reply src/main/java/seedu/address/logic/parser/DemeritCommandParser.java]: @pavithra-achar ya good catch — i initially added a small local helper because i wanted the fix to stay super localized inside DemeritCommandParser first, especially since i was trying to keep the freeze-period change tiny and low-risk.

but after seeing it more carefully, I agree with what you said! ii ll change now!

Comment (kenghow) [reply src/main/java/seedu/address/logic/parser/DemeritCommandParser.java]: @pavithra-achar what do u think now?

Comment (kenghow) [reply src/main/java/seedu/address/logic/parser/DemeritCommandParser.java]: I pretty sure it's correct tho not 100% but I have to merge first for the continuity of the next issue. if really got any issue pls lmk then I can revert back!
